whelk-core/src/main/groovy/whelk/filter/LinkFinder.groovy

```diff
@@ -114,11 +114,19 @@ class LinkFinder {
         items[0][JsonLd.THING_KEY]['@id'] = items[1]['@id']
     }
 
-    private void replaceSameAsLinksWithPrimaries(Map data) {
+    private void replaceSameAsLinksWithPrimaries(Map data, List path = []) {
         // If this is a link (an object containing _only_ an id)
+
         String id = data.get("@id")
         if (id != null && data.keySet().size() == 1) {
-            String primaryId = lookupPrimaryId(id)
+            // Path to same form as in lddb__dependencies.relation
+            String normalizedPath = (path.take(2) == Document.recordPath
+                    ? [JsonLd.RECORD_KEY] + path.drop(2)
+                    : (path.take(2) == Document.thingPath ? path.drop(2) : path)
+            )
+                    .findAll { it instanceof String }
+                    .join('.')
+            String primaryId = lookupPrimaryId(id, normalizedPath)
```

Comment:
Just so I can grasp the context; what was failing here before?

Reply:
Nothing was failing before, however `lookupPrimaryId()` would throw `LinkValidationException` regardless of which relation links to a deleted record.

My previous attempts at https://jira.kb.se/browse/LXL-4209 only fixed the opposite case, i.e. enabling deletion of records with _incoming_ links via certain relations. This new fix is for allowing to delete/save a record with _outgoing_ links to deleted records via these relations.

Reply:
Example:
`<WorkRecord> :derivedFrom <InstanceRecord1>, <InstanceRecord2>`
If <InstanceRecord2> is deleted, then you wouldn't be able to delete or resave <WorkRecord> due to the broken link.

Reply:
Ah; perfect; now I see it! Good work!